app/src/main/java/waiter/EchoServer.java

```diff
@@ -11,8 +11,8 @@ record EchoServer(Reportable reportable) {
 
     public void start(int port) {
 
-        try {
-            reportable.communicate(new Reactor(new ServerSocket(port)));
+        try (ServerSocket serverSocket = new ServerSocket(port)) {
```

Comment:
What is the difference between doing it this way versus having it instantiated in the try block?

Reply:
When instantiated right next to the try keyword in parenthesis, it signifies it is a try with resources, so so long as the object instantiated in the try statement implements AutoClosable (has a close method, which thankfully the ServerSocket does), the objects clean up is taken care of after execution is finished via the try with resources automatically calling the objects close method.

Reply:
ah I see, thanks for clarifying!